### PR TITLE
KTOR-1094 Drop old deprecations

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/CIOEngine.kt [hunk 1]
```diff
@@ -137,6 +137,7 @@ internal class CIOEngine(
 @Suppress("KDocMissingDocumentation")
 @Deprecated(
     "Use ClientEngineClosedException instead",
+    level = DeprecationLevel.ERROR,
```
Comment (rsinukov) [inline ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/CIOEngine.kt:140]: can be an error in 1.6.0 and removed in 2.0.0

File: ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/CIOEngineConfig.kt [hunk 2]
```diff
@@ -79,6 +79,7 @@ public class EndpointConfig {
      */
     @Deprecated(
         "This is deprecated due to the misleading name. Use connectAttempts instead.",
+        level = DeprecationLevel.ERROR,
```
Comment (rsinukov) [inline ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/CIOEngineConfig.kt:82]: same here

File: ktor-client/ktor-client-core/common/src/io/ktor/client/call/HttpClientCall.kt [hunk 3]
```diff
@@ -112,8 +112,8 @@ public open class HttpClientCall internal constructor(
          * Example: [WebSocketSession]
          */
         @Deprecated(
-            "This is going to be removed. " +
-                "Please file a ticket with clarification why and what for do you need it."
+            "This is going to be removed. Please file a ticket with clarification why and what for do you need it.",
```
Comment (rsinukov) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/call/HttpClientCall.kt:115]: same here

**Review by rsinukov — APPROVED** [review]
LGTM, but a lot of deprecation level warning to error changes can be made in 1.6.0 and deleted in 2.0.0

File: ktor-client/ktor-client-core/common/src/io/ktor/client/call/HttpClientCall.kt (hunk 3 above)
Comment (rsinukov) [reply ktor-client/ktor-client-core/common/src/io/ktor/client/call/HttpClientCall.kt:115]: and in a lot of other places

File: ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/CIOEngine.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/CIOEngine.kt:140]: But would it be breaking change?

**Pull request merged**